### ISSUE TITLE
differences in Doxygen output between v1.8.5 and v1.8.14

### ISSUE THREAD

**Issue by jbathegit** [NOAA-EMC/NCEPLIBS-bufr#184]
The [current documentation for this library](https://noaa-emc.github.io/NCEPLIBS-bufr/index.html) is built on WCOSS1 using Doxygen v1.8.5.  But as I'm transitioning over to WCOSS2, which has Doxygen v1.8.14 installed, I'm noticing some unexpected differences.

For an example, compare https://emc.ncep.noaa.gov/emc/pages/infrastructure/bufrlib_jeff/html_mars/ufbint_8f.html (built on WCOSS1 with v1.8.5) against https://emc.ncep.noaa.gov/emc/pages/infrastructure/bufrlib_jeff/html_dogwood/ufbint_8f.html (built on WCOSS2 with v1.8.14).  Specifically, note that in the latter version it no longer generates automatic links to referenced subprograms (e.g. openbf(), ufbseq(), readlc()) within the discussion text, and it no longer includes any cross-referencing (e.g. "References", "Referenced by") links at the bottom of the documentation.

IMHO, these are two big steps backwards for the documentation in terms of functionality and useability.  Has anyone else in @NOAA-EMC/nceplibs seen this or have an idea how to fix it?  I don't see anything obvious in the NCEPLIBS-bufr Doxyfile configuration settings nor based on a quick issue search online.

**Comment by kgerheiser** [comment]
That is curious. I tried to search the cause of this, but couldn't find anything. I agree, it's nice how the first one is formatted with the references.

**Comment by edwardhartnett** [comment]
I am using references and referenced by everywhere with 1.8.16 and that seems to work fine.

**Comment by jbathegit** [comment]
@edwardhartnett where do you have access to Doxygen 1.8.16?

I'm now wondering if maybe there's something missing or misconfigured in the installation of 1.8.14 on WCOSS2, b/c otherwise I can't imagine someone else in the wider user community wouldn't have noticed this by now if it were an issue with Doxygen itself.

**Comment by edwardhartnett** [comment]
My ubuntu system has doxygen 9.1, but I installed 1.8.17/18 for testing a bug that came up recently. I believe 1.8.17 is what is provided when we build on GitHub actions - are you doing that?

**Comment by kgerheiser** [comment]
On my local machine with Doxygen 1.9.2 the output looks like this:

<img width="1051" alt="Screen Shot 2022-02-02 at 13 05 55" src="https://user-images.githubusercontent.com/3209794/152221108-13b2bf93-0862-4521-b580-0bb6df4c3313.png">

It has the references, but not referenced by.

**Comment by jbathegit** [comment]
> My ubuntu system has doxygen 9.1, but I installed 1.8.17/18 for testing a bug that came up recently. I believe 1.8.17 is what is provided when we build on GitHub actions - are you doing that?

I've just been using whatever version of Doxygen is provided on the machines where I've been doing my own testing, so basically WCOSS1 and WCOSS2.  I haven't tried looking at any of the docs generated from GitHub actions, so maybe that's an option.  However, that would also mean that I can't test Doxygen-related changes or updates without first pushing a commit up to GitHub.

**Comment by jbathegit** [comment]
Interesting that 1.8.16 seems to have both references and referenced by, but 1.9.2 seems to be missing the latter.  Not sure what to make of that.

What about the automatic links to referenced subprograms (e.g. openbf(), ufbseq(), readlc()) within the discussion text - do you see those being generated in your output, like what's shown in my above 1.8.5 example of ufbint from WCOSS1?

**Comment by jbathegit** [comment]
Since nothing further has transpired on this issue, and since I'm getting ready to do another (mini)release (see #227), I looked into other options.

My office Linux system (emc-lw7-jator) has Doxygen v1.8.5, but it doesn't have CMake 3.15+ installed so I couldn't build NCEPLIBS-bufr there.

After looking into other GFE options, what I ended up doing was just installing Doxygen v1.9.4 in the Cygwin environment on my personal Windows PC at home, since it also has the requisite versions of CMake, GNU compilers and other stuff needed to build NCEPLIBS-bufr there.  It's a bit of a kludge to hop the Doxygen HTML output from there back thru the VPN and WinSCP to emcrzdm, but it's doable and the output looks reasonable, so in the absence of any other options I'll just go with that.
